src/sentry/filestore/gcs.py

```diff
@@ -361,3 +366,43 @@ def get_available_name(self, name, max_length=None):
             name = clean_name(name)
             return name
         return super().get_available_name(name, max_length)
+
+    def try_del(self, callable: Callable[[], None]) -> None:
+        self._try(callable)
+
+    def try_get(self, callable: Callable[[], None]) -> None:
+        self._try(callable)
+
+    def try_set(self, callable: Callable[[], None]) -> None:
+        self._try(callable)
+
+    def _try(self, callable: Callable[[], None]) -> None:
+        # The default policy since 2018 has been to retry five times in a loop with no backoff
+        # for gets, sets, and deletes. This behavior has been preserved here.
+        #
+        # To implement a custom retry policy see `src/sentry/utils/retries.py` and the
+        # `GoogleCloudStorageReplayUploadPolicy` class.
+        try_repeated(callable)
+
+
+class GoogleCloudStorageReplayUploadPolicy(GoogleCloudStorage):
```

Comment:
```suggestion
class GoogleCloudStorageWithReplayUploadPolicy(GoogleCloudStorage):
```
The current name could be confused as a policy class, rather than storage